flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -247,7 +253,10 @@ public CatalogDatabase getDatabase(String databaseName) throws DatabaseNotExistE
 
 		Map<String, String> properties = hiveDatabase.getParameters();
 
-		properties.put(HiveCatalogConfig.DATABASE_LOCATION_URI, hiveDatabase.getLocationUri());
+		boolean isGeneric = getObjectIsGeneric(properties);
+		if (!isGeneric) {
+			properties.put(SqlCreateHiveDatabase.DATABASE_LOCATION_URI, hiveDatabase.getLocationUri());
+		}
```

Comment:
I put the property keys in SqlNode because the SqlNode has to add these extra properties for extended Hive semantics.

Reply:
But generally, the Catalog code should not see the `SqlNodes`, because the former belongs to the planner, the later belongs to parser.

Reply:
Both SqlNode and HiveCatalog need to access these properties. I can't put them in the hive connector because hive connector depends on `sql-parser` and not the other way around. So any suggestions where they should go?

Reply:
After another think i think it is okey, because we passes around the Hive specific properties through the `SqlNode` which is the root cause i think.

We can have a refactor when we really have a new planner.